pkg/virt-controller/watch/vm/vm.go

```diff
@@ -1621,10 +1621,32 @@ func syncVolumeMigration(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineIn
 	}
 	vmCond := controller.NewVirtualMachineConditionManager()
 	vmiCond := controller.NewVirtualMachineInstanceConditionManager()
-	// Something went wrong with the VMI while the volume migration was in progress
-	if vmi == nil && vmCond.HasConditionWithStatus(vm, virtv1.VirtualMachineConditionType(virtv1.VirtualMachineInstanceVolumesChange), k8score.ConditionTrue) {
+
+	srcMigVols := make(map[string]string)
+	for _, v := range vm.Status.VolumeUpdateState.VolumeMigrationState.MigratedVolumes {
```

Comment:
maybe it needs a simple comment, "// Checks if VM volumes have been recovered to their original PVCs after a failed migration. "